### PR TITLE
Implement initial version of calendar utility

### PR DESCRIPTION
**Pull request by AymarMuhikira**
-Call function to populate calendar panel with events
-Servlet gets the credential and gets the list of events from utility
-Utility makes API call to get calendar service and events.

### CODE REVIEW

File: src/main/java/com/google/sps/utility/CalendarUtility.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.Event;
+import java.io.IOException;
+import java.util.List;
+
+public class CalendarUtility {
```
Comment (austinteshuba) [inline src/main/java/com/google/sps/utility/CalendarUtility.java:27]: Public classes require [Javadoc](https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required)

File: src/main/java/com/google/sps/utility/CalendarUtility.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  private CalendarUtility() {}
+
+  public static Calendar getCalendarService(Credential credential) {
```
Comment (austinteshuba) [inline src/main/java/com/google/sps/utility/CalendarUtility.java:31]: Public functions require [Javadoc](https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required)

File: src/main/java/com/google/sps/utility/CalendarUtility.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    HttpTransport httpTransport = UrlFetchTransport.getDefaultInstance();
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    String applicationName = AuthenticationUtility.APPLICATION_NAME;
+
+    return new Calendar.Builder(httpTransport, jsonFactory, credential)
+        .setApplicationName(applicationName)
+        .build();
+  }
+
+  public static List<Event> getCalendarEvents(Calendar calendarService) throws IOException {
```
Comment (austinteshuba) [inline src/main/java/com/google/sps/utility/CalendarUtility.java:41]: Public functions require [Javadoc](https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required)

File: src/main/webapp/populateData.js [hunk 4]
```diff
@@ -89,3 +89,28 @@ function populateTasks() {
...
+/**
+ * Populate Calendar container with user's events
+ */
+function populateCalendar() {
+  const calendarContainer = document.querySelector('#calendar');
+  fetch('/calendar')
+      .then((response) => {
+        // If response is a 403, user is not authenticated
+        if (response.status === 403) {
+          throw new Error();
```
Comment (austinteshuba) [inline src/main/webapp/populateData.js:101]: Use the new AuthenticationError that I defined in pr #21 (it's new so you likely didn't see this yet). 

Then, you can actually check if the thrown error is an instance of AuthenticationError and only sign out in that case

**Comment by AymarMuhikira** [comment]
@googlebot I fixed it.

**Pull request closed without merging**